src/lib/viewers/controls/media/MediaSettingsControls/MediaSettingsMenu.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import classNames from 'classnames';
+import { decodeKeydown } from '../../../../util';
+import MediaSettingsContext, { Menu } from './MediaSettingsContext';
+import './MediaSettingsMenu.scss';
+
+export type Props = React.PropsWithChildren<{
+    className?: string;
+    name: Menu;
+}>;
+
+export default function MediaSettingsMenu({ children, className, name }: Props): JSX.Element | null {
+    const [activeIndex, setActiveIndex] = React.useState(0);
+    const { activeMenu, setActiveRect } = React.useContext(MediaSettingsContext);
+    const isActive = activeMenu === name;
+    const menuElRef = React.useRef<HTMLDivElement>(null);
+    const menuItemElRef = React.useRef<HTMLDivElement>(null);
+
+    const handleKeyDown = (event: React.KeyboardEvent): void => {
+        const key = decodeKeydown(event);
+        const max = React.Children.toArray(children).length - 1;
+
+        if (key === 'ArrowUp' && activeIndex - 1 >= 0) {
+            setActiveIndex(activeIndex - 1);
+        }
+
+        if (key === 'ArrowDown' && activeIndex + 1 <= max) {
+            setActiveIndex(activeIndex + 1);
+        }
+    };
+
+    React.useEffect(() => {
+        const { current: menuEl } = menuElRef;
+
+        if (menuEl && isActive) {
+            setActiveRect(menuEl.getBoundingClientRect());
+        }
+    }, [isActive, setActiveRect]);
+
+    React.useEffect(() => {
+        const { current: menuItemEl } = menuItemElRef;
+
+        if (menuItemEl && isActive) {
+            menuItemEl.focus();
+        }
+    }, [activeIndex, isActive]);
```

Comment:
Is `activeIndex` included here to trigger the effect when the `menuItemElRef` changes? If so, maybe a comment would be helpful here